src/main/java/duke/core/TypoCorrector.java

```diff
@@ -0,0 +1,110 @@
+package duke.core;
+
+import org.apache.commons.text.similarity.LevenshteinDistance;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * This is a command typo corrector for Duke user command.
+ * It provides a method TypoCorrector.CommandCorrection which takes in an invalid input command
+ * and return a possible matched command.
+ *
+ * @author  HUANG XUAN KUN
+ * @version 1.2
+ */
+public class TypoCorrector {
+
+    //The maximum ratio changes of a text in % that is acceptable
+    private final static double MAX_DISTANCE_DIFF_RATIO = 0.5;
+
+
+    //Sets of "Dictionaries" for the command matching, categorised by number of keywords contain in a supported commands.
+    private final static ArrayList<String> oneKeywordCommand = new ArrayList<String>(Arrays.asList("bye"));
+    private final static ArrayList<String> twoKeywordsCommands = new ArrayList<String>(Arrays.asList("list patients", "list tasks"));
+    private final static ArrayList<String> otherCommands = new ArrayList<String>(Arrays.asList("update patient", "update command", "delete patient", "delete task", "add task", "add patient", "assign by"));
+
+    /**
+     * This method take in an user input command with typo and return a possible matches
+     * If the return string is equal to the input command, there is no match.
+     *
+     * @param command the full userInput command without parsing
+     * @return
+     */
+    public static String CommandCorrection(String command) {
+        String[] splitCommand = command.split("\\s+");
+        int commandSize = splitCommand.length;
+        String firstKeyword = splitCommand[0].toLowerCase();
+        String closestMatch;
+        if (commandSize == 1 || command.length() <= 6) {
+            closestMatch = matchStringFromDict(command, oneKeywordCommand);
+            if (isSimilar(command, closestMatch)){
+                return closestMatch;
+            }
+        } else if (commandSize == 2 || command.length() <= 16) {
```

Comment:
Removing this check, or minimizing its length is likely the best approach. Checking for an exact length of an array is already fairly thorough, so checking the character length seems like it may not be necessary, honestly.

Reply:
Thanks! The fix is implemented. Do review it again.

Reply:
@lmtaek  Thanks for the catch. The fix is implemented with passed checks. Do review it again